src/QMCWaveFunctions/Fermion/DiracDeterminantBatched.cpp

```diff
@@ -46,38 +47,129 @@ void DiracDeterminantBatched<DET_ENGINE_TYPE>::set(int first, int nel, int delay
+template<typename DET_ENGINE>
+void DiracDeterminantBatched<DET_ENGINE>::invertPsiM(DiracDeterminantBatchedMultiWalkerResource<DET_ENGINE>& mw_res,
+                                                     OffloadPinnedValueMatrix_t& logdetT)
 {
   ScopedTimer inverse_timer(InverseTimer);
-  det_engine_.invert_transpose(logdetT, LogValue);
+  det_engine_.invert_transpose(logdetT, mw_res.log_values);
+  LogValue = mw_res.log_values[0];
+
+#ifndef NDEBUG
+  auto& engine_psiMinv = det_engine_.get_nonconst_psiMinv();
+  dummy_vmt.attachReference(engine_psiMinv.data(), engine_psiMinv.rows(), engine_psiMinv.cols());
+#endif
+
 }
 
-template<typename DET_ENGINE_TYPE>
-void DiracDeterminantBatched<DET_ENGINE_TYPE>::mw_invertPsiM(const RefVectorWithLeader<WaveFunctionComponent>& wfc_list,
-                                                             const RefVector<const ValueMatrix_t>& logdetT_list)
+template<typename DET_ENGINE>
+void DiracDeterminantBatched<DET_ENGINE>::mw_invertPsiM(const RefVectorWithLeader<WaveFunctionComponent>& wfc_list,
+                                                        const RefVector<const OffloadPinnedValueMatrix_t>& logdetT_list)
 {
-  auto& wfc_leader = wfc_list.getCastedLeader<DiracDeterminantBatched<DET_ENGINE_TYPE>>();
+  auto& wfc_leader = wfc_list.getCastedLeader<DiracDeterminantBatched<DET_ENGINE>>();
   ScopedTimer inverse_timer(wfc_leader.InverseTimer);
   const auto nw = wfc_list.size();
+  // It would be nice to not make this call over and over.
+  wfc_leader.mw_res_->log_values.resize(nw);
 
-  RefVectorWithLeader<DET_ENGINE_TYPE> engine_list(wfc_leader.det_engine_);
-  RefVector<LogValueType> log_value_list;
+  RefVectorWithLeader<DET_ENGINE> engine_list(wfc_leader.det_engine_);
   engine_list.reserve(nw);
 
   for (int iw = 0; iw < nw; iw++)
   {
-    auto& det = wfc_list.getCastedElement<DiracDeterminantBatched<DET_ENGINE_TYPE>>(iw);
-    engine_list.push_back(det.det_engine_);
-    log_value_list.push_back(det.LogValue);
+    auto& det = wfc_list.getCastedElement<DiracDeterminantBatched<DET_ENGINE>>(iw);
+    engine_list.push_back(det.get_det_engine());
+    wfc_leader.mw_res_->log_values[iw] = {0.0,0.0};
+    //typename decltype(mw_res_)::element_type{}; // Logdet.LogValue;
+  }
+
+  DET_ENGINE::mw_invertTranspose(engine_list, logdetT_list, wfc_leader.mw_res_->log_values);
+  for (int iw = 0; iw < nw; ++iw)
+  {
+    auto& det    = wfc_list.getCastedElement<DiracDeterminantBatched<DET_ENGINE>>(iw);
+    det.LogValue = wfc_leader.mw_res_->log_values[iw];
   }
 
-  DET_ENGINE_TYPE::mw_invert_transpose(engine_list, logdetT_list, log_value_list);
+#ifndef NDEBUG
+  for (int iw = 0; iw < nw; ++iw)
+  {
+    auto& det            = wfc_list.getCastedElement<DiracDeterminantBatched<DET_ENGINE>>(iw);
+    auto& engine_psiMinv = det.det_engine_.get_nonconst_psiMinv();
+    det.dummy_vmt.attachReference(engine_psiMinv.data(), engine_psiMinv.rows(), engine_psiMinv.cols());
+  }
+#endif
+}
+
+template<>
```

Comment:
Not getting this point. Dropping the filter needs to be consist with the change how psiMinv is stored. So it is a decision inside DET_ENGINE, I don't think adding DiracDeterminantDetails is necessary.

Reply:
I'd already gotten mw_recompute working with the unfiltered lists and wanted move to using only contiguous memory for the batched inversion when the filtered lists being handed to the DET_ENGINE from DiracBatchedDeterminant  was changed in your PR. I wanted to maintain API and walker ref list to memory resource relationship I had been working with.  I'll review this. If we eliminate the DiracDeterminantBase we won't need this either.

Reply:
It is a bit tricky because to use unfiltered list, the SPO evaluation needs to be fast enough, we are not there yet. So it is better to use the filtered list and only compute SPO and inversion for them in order not to hurt to much in performance.

Reply:
Also we potentially need to support slow-to-evaluate SPO, so filtered list is good.

Reply:
SPO evaluation should be very fast on the accelerator. The main issue with SPO evaluation is the data movement. I think on the accelerator leaving the layout of the walkers in device memory the same for an entire block is worth pursuing. If we hand the compute API partial walkers lists but want to use a shared and persistent memory layout this is a mess. 

It would make more sense to pass the execute masks all the way down through the API.  The filtered list is only convenient if you are terminating at single walker computes with per walker pointers to pointers. This only works nicely if you mix up the concept of computation method and data objects as the current DiracMatrix does. Doing this means that hoop jumping to allocated SoA style data will always be convoluted since the code always expects to find data in 1 walker compute objects instead of contiguous vectors for multiple walkers that have straight forward transfer semantics to and from the device and if we need to narrow or widen precision between those buffers on the the device.

It's always going to be much slower to make many calls to the async transfer engine for single walker data elements than for contiguous many walker element arrays. And I think this is frequently the important bottle neck.

I think we need some real benchmarks that reflect the impact of this.

Reply:
`DistanceTableData.h` mw_recompute() API carries a full list and masks.
SoaDistanceTableAB.h runs walker by walker. So can benefit from masks.
SoaDistanceTableABOMPTarget.h runs walker batched because a fused memory buffer and thus imply ignores the mask.